helm-chart/renku/templates/graph/db-encryption-secret.yaml

```diff
@@ -0,0 +1,60 @@
+---
+{{- $tokenRepository := default (randAlphaNum 32) .Values.graph.tokenRepository.aesEncryptionKey | b64enc | quote -}}
+{{- $webhookAesEncryptionKey := default (randAlphaNum 32) .Values.graph.webhookService.aesEncryptionKey | b64enc | quote -}}
+
+{{/* If this secret already exist just re-use it */}}
+{{- $secretName := include "renku.graph.dbEncryptionSecret.name" . -}}
+{{- $graphEncKeys := lookup "v1" "Secret" .Release.Namespace $secretName -}}
+{{- if $graphEncKeys -}}
+  {{- $tokenRepository = index $graphEncKeys.data "tokenRepository-aes-encryption-key" -}}
+  {{- $webhookAesEncryptionKey = index $graphEncKeys.data "webhook-aes-encryption-key" -}}
+{{/* If it doesn't exist check if it's in the previous form or in the values */}}
+{{- else -}}
+  {{- $secretName := include "renku.graph.tokenRepository.fullname" . -}}
+  {{- $oldTokenSecret := lookup "v1" "Secret" .Release.Namespace $secretName -}}
+  {{- if $oldTokenSecret -}}
+    {{- $newTokenSecret :=  index $oldTokenSecret.data "tokenRepository-tokenEncryption-secret" | b64dec | b64dec | trim -}}
+    {{/* If a secret has been specified in the values make sure that corresponds to the base64-decoded secret already present in the cluster, or fail */}}
+    {{- if and .Values.graph.tokenRepository.aesEncryptionKey (ne $newTokenSecret .Values.graph.tokenRepository.aesEncryptionKey) -}}
+       {{- fail (printf "The token repository AES encryption key provided in `graph.tokenRepository.aesEncryptionKey` does not match with the base64 decoded key present in %s. %s != %s" $secretName $newTokenSecret (.Values.graph.tokenRepository.aesEncryptionKey | b64enc)) -}}
+    {{- else -}}
+      {{- $tokenRepository = $newTokenSecret | b64enc | quote -}}
+    {{- end -}}
+  {{- end -}}
+
+  {{- $secretName := include "renku.graph.webhookService.fullname" . -}}
+  {{- $oldWebhookSecret := (lookup "v1" "Secret" .Release.Namespace $secretName) -}}
+  {{- if $oldWebhookSecret -}}
+    {{- $newWebhookSecret := index $oldWebhookSecret.data "webhookService-hookToken-secret" | b64dec | b64dec | trim -}}
+    {{/* If a secret has been specified in the values make sure that corresponds to the base64-decoded secret already present in the cluster, or fail */}}
+    {{- if and .Values.graph.webhookService.aesEncryptionKey (ne $newWebhookSecret .Values.graph.webhookService.aesEncryptionKey) -}}
+      {{- fail (printf "The webhook AES encryption key provided in `.graph.webhookService.aesEncryptionKey` does not match with the base64 decoded key present in %s." $secretName) -}}
```

Comment:
Why not have the same message here as what you have above? In the message above you print the different secret values.

Reply:
Typo. Fixed, thanks.